sklearn/preprocessing/data.py

```diff
@@ -148,15 +154,21 @@ def scale(X, axis=0, with_mean=True, with_std=True, copy=True):
             raise ValueError("Can only scale sparse matrix on axis=0, "
                              " got axis=%d" % axis)
         if with_std:
+            if with_std not in (1, 2, True):
+                raise ValueError("Invalid value for `with_std`: {}".format(
+                                 str(with_std)))
             _, var = mean_variance_axis(X, axis=0)
             var = _handle_zeros_in_scale(var, copy=False)
-            inplace_column_scale(X, 1 / np.sqrt(var))
+            inplace_column_scale(X, 1 / (with_std * np.sqrt(var)))
     else:
         X = np.asarray(X)
         if with_mean:
             mean_ = np.nanmean(X, axis)
         if with_std:
```

Comment:
Is this check still needed if we have `if with_std not in (...)`?

Reply:
@thomasjpfan Yes, so we can restrict the user to (1, 2, True) values.

Reply:
Yes, it's an additional check.